### ISSUE TITLE
Required header yaml.h not found

### ISSUE THREAD

**Issue by kevindugan** [ubccr/hpc-toolset-tutorial#174]
I was working through this tutorial recently and ran into issues within the OnDemand section. Some ruby gems were not able to compile because a `yaml.h` header was missing. 

I was able to work around the issue by making the following change

```diff
diff --git a/ondemand/install.sh b/ondemand/install.sh
index 546e5ac..f0371bf 100755
--- a/ondemand/install.sh
+++ b/ondemand/install.sh
@@ -79,6 +79,7 @@ log_info "Generating new httpd24 and dex configs.."
 log_info "Adding new theme to dex"
 sed -i "s/theme: ondemand/theme: hpc-coop/g" /etc/ood/dex/config.yaml

+dnf --enablerepo=powertools install -y libyaml-devel
 dnf clean all
 rm -rf /var/cache/dnf

```

and forcing the ondemand image be built locally instead of pulling from dockerhub. However, I'm not familiar enough with this package to know if this is the best place for the libyaml-devel of if it would be better suited in one of the base images. If this is an appropriate change, I'd be happy to submit a merge request.

BTW, this was run on an Apple M1 Pro.

**Comment by johrstrom** [comment]
Hi thanks for the reminder! I've created a pull request for the same. Thanks!

**Comment by kevindugan** [comment]
Aaaaaaand I just found that the powertools repo is enabled in the slurm image. Disregard my previous comment 😄

**Comment by widyono-cets** [comment]
Could someone rebuild the image in the repo with this fix? ubccr/hpcts:ondemand-2023.07 doesn't have libyaml-devel addition.
